### PR TITLE
chore: declare required extensions for JetBrains IDE.

### PR DESCRIPTION
**Pull request by themr0c**
chore: declare required extensions for JetBrains IDE.

See:  https://www.jetbrains.com/help/pycharm/managing-plugins.html#required-plugins

- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).

To test: https://workspaces.openshift.com/#https://github.com/themr0c/che-docs/tree/chore-jetbrains-required-plugins?che-editor=che-incubator/che-pycharm/latest doesn't work out of the box: we need also to specify the registryURL.

### CODE REVIEW

**Comment by themr0c** [comment]
https://workspaces.openshift.com/#https://github.com/themr0c/che-docs/tree/chore-jetbrains-required-plugins?che-editor=che-incubator/che-pycharm/latest fails, because Dev Spaces registry doesn't include PyCharm. 
There is no documented URL parameter to use an editor outside the default registry,

Falling back to using the `.che/che-editor.yaml` file to set the IDE.

The workspace fails to start with `exceeded quota` error: 

```
Danger alert:Detected unrecoverable deployment condition: FailedCreate pods "workspace1ad6a39c512a4fe2-75b9dd9d66-5rm6k" is forbidden: exceeded quota: for-ffloreth-compute, requested: limits.memory=8448Mi, used: limits.memory=0, limited: limits.memory=7Gi
```

**Comment by azatsarynnyy** [comment]
FYI: It's also possible to provide a link to an editor Devfile as a value for the che-editor URL parameter.
E.g.: `?che-editor=https://gist.githubusercontent.com/azatsarynnyy/656942e221114befacf7970c619288df/raw/ef047bda4f77574bb1f214f31af39f66b5beb146/devfile.yaml`

**Comment by themr0c** [comment]
I removed the commit to set PyCharm as default.

The pull as it is is useful even if it cannot be tested on Che.

File: .idea/externalDependencies.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    <plugin id="com.intellij.tasks" />
+    <plugin id="com.jetbrains.plugins.ini4idea" />
+    <plugin id="com.jetbrains.sh" />
+    <plugin id="mobi.hsz.idea.gitignore" />
+    <plugin id="org.asciidoctor.intellij.asciidoc" />
+    <plugin id="org.jetbrains.plugins.github" />
+    <plugin id="org.jetbrains.plugins.terminal" />
+    <plugin id="org.jetbrains.plugins.yaml" />
+  </component>
+</project>
```
Comment (themr0c) [inline .idea/externalDependencies.xml]:
```suggestion
</project>

```

**Review by azatsarynnyy — APPROVED** [review]

**Comment by themr0c** [comment]
But why is the link checker failing on https://che.eclipseprojects.io/? It doesn't make sense.

**Pull request merged**